src/test/java/com/peel/okhttp3/staleiferror/StaleIfErrorInterceptorTest.java

```diff
@@ -15,14 +15,310 @@
+				.cache(new Cache(new File("./caches"), (long) (1024 * 1024 * 10)))
+				.connectTimeout(2, TimeUnit.SECONDS)
+				.build();
+		
+		clientWith4SecMaxStale = new OkHttpClient.Builder()
+				.addInterceptor(new StaleIfErrorInterceptor(4, TimeUnit.SECONDS))
+				.sslSocketFactory(handshakeCertificates.sslSocketFactory(), handshakeCertificates.trustManager())
+				.cache(new Cache(new File("./caches"), (long) (1024 * 1024 * 10)))
+				.connectTimeout(2, TimeUnit.SECONDS)
+				.build();
+
+		url = server.url("/");
+	}
+
+	private Request.Builder request() {
+		return new Request.Builder().url(url);
+	}
+
+	@Test
+	public void testCachingWithMaxAge() {
+		//setup a server with mock response and max age 2 sec
+		MockResponse mockResponse = new MockResponse().setResponseCode(200)
+				.setBody("Hello from mock server")
+				.addHeader("Content-Type: text/html; charset=utf-8")
+				.addHeader("Cache-Control", "public, max-age=2");
+		MockResponse mock404Response = new MockResponse().setResponseCode(404)
+				.addHeader("Content-Type: text/html; charset=utf-8")
+				.addHeader("Cache-Control", "public, max-age=2");
+		server.enqueue(mockResponse);
+		server.enqueue(mock404Response);
+		server.enqueue(mockResponse);
+		
+		Request request = request().get().build();
+		Response response = null;
+		try {
+			//make first call and cache the response
+			response = client.newCall(request).execute();
+			assertNotNull(response);
+			assertNull(response.cacheResponse());
+			assertNotNull(response.networkResponse());
+			assertEquals(200, response.code());
+			response.body().close();
+			
+			//get response from the local cache.
+			response = client.newCall(request).execute();
+			assertNotNull(response);
+			assertNotNull(response.cacheResponse());
+			assertNull(response.networkResponse());
+			assertEquals(200, response.code());
+			response.body().close();
+			
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+			
+			//get response from the local cache 
+			response = client.newCall(request).execute();
+			assertNotNull(response);
+			assertNotNull(response.cacheResponse());
+			assertNull(response.networkResponse());
+			assertEquals(200, response.code());
+			response.body().close();
+			try {
+				Thread.sleep(5000);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+			
+			//get response from the stale cache
+			response = client.newCall(request).execute();
+			assertNotNull(response);
+			assertNotNull(response.cacheResponse());
+			assertNull(response.networkResponse());
+			assertEquals(200, response.code());
+			response.body().close();
+			
+			//bring back the server here
+			response = client.newCall(request).execute();
+			assertNotNull(response);
+			assertNull(response.cacheResponse());
+			assertNotNull(response.networkResponse());
+			response.body().close();
+			
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
why do we catch exception?

Reply:
we dont need it, I will remove the try catch blocks.